src/shared/gravity_value.c

```diff
@@ -2258,12 +2351,23 @@ inline gravity_string_t *gravity_string_new (gravity_vm *vm, char *s, uint32_t l
     obj->s = (char *)s;
     obj->len = len;
     obj->alloc = (alloc) ? alloc : len;
+    obj->s = mem_alloc(NULL, obj->alloc);
+    strcpy(obj->s, s);
     if (s && len) obj->hash = gravity_hash_compute_buffer((const char *)s, len);
 
     if (vm) gravity_vm_transfer(vm, (gravity_object_t*) obj);
     return obj;
 }
 
+inline void gravity_string_concat_cstring (gravity_vm *vm, gravity_string_t *obj, char *cstring) {
+    uint32_t len = (uint32_t)(strlen(obj->s) + strlen(cstring));
+    uint32_t alloc = MAXNUM(len+1, DEFAULT_MINSTRING_SIZE);
+    obj->s = mem_realloc(NULL, obj->s, alloc);
```

Comment:
Instead of always perform a men_realloc you should first check if (obj->alloc - obj->len) is big enough